test/gtest/log.cpp

```diff
@@ -29,8 +29,8 @@
 
 #include <miopen/config.h>
 #include <miopen/fusion_plan.hpp>
-#include <stdlib.h>
-#include <random>
+#include <cstdlib>
+#include "random.hpp"
```

Comment:
minor: perhaps we should name (and move) our "random.hpp" to "miopen/random.hpp" or if that's not feasible then "miopen_random.hpp" to avoid confusion with `<random>` from STL.

Reply:
@amberhassaan
> perhaps we should name (and move) our "random.hpp" to "miopen/random.hpp" 

The library does not use random.hpp (while tests and driver use different versions of this file). Therefore, this is not possible and not needed.

> or if that's not feasible then "miopen_random.hpp" to avoid confusion with <random> from STL.

But I do not see any confusion. For example, double quotes imply that the included file is not from the standard library.

Reply:
You're right `"random.hpp"` should convey that it's local. However, we are not consistent in our repository and we often use `<>` for local MIOpen includes.

Reply:
The quoted form should only be used for headers that are relative to the including file, and everything else should use the angle brackets. In general, MIOpen follows this rule. This is also the guideline from C++ Core Guidelines [SF.12](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-incform).

Reply:
@CAHEK7 Please resolve https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2397#discussion_r1368869376

Reply:
@CAHEK7 "../random.hpp" here is expected to comply the rule.

Reply:
@atamazov @pfultz2 done.
But what about the others includes? Mixing different paradigms of including headers may be confusing.
```CPP
#include "log.hpp"
#include "tensor_util.hpp"
#include "get_handle.hpp"
...
#include "../random.hpp"
```
Shall everything be fixed uniformly across the project, because it's a very common case where we include files from the project using quotes and implying relative includes.

Almost all the files in `gtests` directory and some of the files in the `test` directory should be fixed.